src/Controls/src/Core/NavigationPage/NavigationPageToolbar.cs

```diff
@@ -273,7 +273,23 @@ void OnPropertyChanged(object sender, System.ComponentModel.PropertyChangedEvent
 
 		Color GetBarTextColor() => _currentNavigationPage?.BarTextColor;
 		Color GetIconColor() => (_currentPage != null) ? NavigationPage.GetIconColor(_currentPage) : null;
-		string GetTitle() => GetTitleView() != null ? String.Empty : _currentPage?.Title;
+
+		string GetTitle()
+		{
+			if (GetTitleView() != null)
+			{
+				return string.Empty;
+			}
+
+			Page target = _currentPage;
+
+			if (_currentNavigationPage?.CurrentPage is TabbedPage tabbedPage && !string.IsNullOrEmpty(tabbedPage.Title))
```

Comment:
I wonder if this needs to check the page type at all.

I'm testing a few variations of pages on Xamarin.Forms, and the Title always seems to just be whatever the current page is on the navigation page. 

![image](https://github.com/dotnet/maui/assets/5375137/6df4c3bc-508f-4d84-8c12-219cd9b4f9df)

Is there ever a scenario where
_currentNavigationPage.CurrentPage.Title isn't what we want here?